### PR TITLE
[WIP] [FEATURE] Add i18n support

### PR DESCRIPTION
**Pull request by kennym**
This PR is going to implement i18n in the EAC dApp so we can reach more people who speak other languages natively.

For i18n in our dApp I propose using https://github.com/alibaba/react-intl-universal . I've researched other i18n libraries like:

- https://github.com/yahoo/react-intl
- https://github.com/i18next/react-i18next/

The reason for choosing `react-intl-universal` boils down to its simpler API, but still covers all basic features and more. Translations are loaded from JSON files, and people can easily contribute new translations in the future. `react-intl` is bloated and complex. `react-i18next` provides an unnecessarily complex API for a decent integration.

The language is detected based on the user language set in the browser, which is determined by `navigator.language` or `navigator.userLanguage`.
The default language is of course English, and in order to specify another language you can provide the parameter `lang`, e.g.: `http://localhost:8080?lang=es-ES`. Current language options are:

- `en-EN`
- `es-ES`

and we can easily add new ones.

TODO:

- [x] Add language selector
- [x] Implement navigator.language-based for loading languages
- [x] Provide a TRANSLATIONS guideline for contributors
- [x] Tests should work
- [ ] Convert all strings to translatables
- [ ] Provide complete Spanish translations
- [ ] Implement cookie-based solution for loading languages
- [ ] Provide a lint hook that checks correct format of translation strings

### CODE REVIEW

File: app/components/App/App.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import _ from 'lodash';
+import http from 'axios';
```
Comment (kosecki123) [inline app/components/App/App.js]: use fetch api instead

File: app/components/App/App.js [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import _ from 'lodash';
```
Comment (kosecki123) [inline app/components/App/App.js]: import `find` function only or rewrite without loadsh

Comment (kennym) [reply app/components/App/App.js]: ok 👍

File: app/components/App/App.js (hunk 1 above)
Comment (kennym) [reply app/components/App/App.js]: ok 👍

File: app/components/App/App.js [hunk 3]
```diff
@@ -43,8 +61,38 @@ class App extends Component {
     }
   }
 
+  /*
+    Load locales and resume page initialisation after that is done.
+  */
+  loadLocales(){
```
[the remark this answers is not in the document]
Comment (kennym) [reply app/components/App/App.js]: ok, thanks! That was my first instinct, but then I saw that other functions are formatted that way.

**Comment by kennym** [comment]
@kosecki123 hi! I'm putting in 15 minutes a day into this, but I figured it may take some time to make all strings translatable. The good thing is that strings can be i18nized and translated over time. I'd recommend merging this, before more merge conflicts occur.

**Comment by Kuzirashi** [comment]
Hello @kennym, we can't consider merging because tests are broken (seems like easy fix):
```
/home/travis/build/chronologic/eth-alarm-clock-dapp/app/components/Common/URLNotFound.js
  16:25  error  Unexpected usage of singlequote  jsx-quotes
```

**Comment by kennym** [comment]
So `intl.get` introduces newlines around the strings, and it feels hacky to accommodate the snapshots to the new output, and I also don't feel it's necessary to fork the library and update the part is we're talking about aesthetics and it's still valid HTML that outputs the same.

@Kuzirashi I wonder how you would go about fixing this? Using regex matches for expectations instead of a snapshot match?

```
               className="fa fa-sitemap"
              />
    -           Active TimeNodes:
    +
    +         Active TimeNodes
    +         :
            </span>
            <span
              className="timenode-count"
            />
          </div>
    @@ -43,11 +45,13 @@
              className="active-timenodes"
            >
              <i
                className="fa fa-th-large"
              />
    -          Current Block Number:
    +
    +         Current Block Number
    +         :
            </span>```

**Comment by Kuzirashi** [comment]
@kennym You can run jest -u, to update snapshots, or yarn test -u. It should fix snapshots automatically, don't worry. :)

**Comment by kennym** [comment]
@Kuzirashi thanks! I'm glad I asked :-)

**Comment by kosecki123** [comment]
I'm in favour of merging current state of the PR, with one change request which is to disable the detection until language packs are more complete. Otherwise people with supported languages will get partially translated dapp without a easy way to pick the language.

**Comment by kennym** [comment]
@kosecki123 agreed - if I add the language selector before this gets merged would we be able to keep the language detection for Spanish at least?

**Comment by kennym** [comment]
@kosecki123 @Kuzirashi I've added the language selector tool. It currently looks like this:

image

File: app/scss/style.scss [hunk 4]
```diff
@@ -292,4 +292,76 @@ hr {
...
+}
+
+.flag-select {
+  position: relative;
+  display: inline-block;
+  vertical-align: inherit;
+  padding-bottom: 5px;
+  color: #4d4d4d;
+  text-align: left; }
+  .flag-select img {
```
Comment (Kuzirashi) [inline app/scss/style.scss]: We're using SCSS, can you make sure this code is SCSS and takes advantage of its features?

File: app/stores/LanguageStore.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import intl from 'react-intl-universal';
+
+const SUPPORTED_LOCALES = [
+  {
+    name: 'English',
+    value: 'en-US',
+    selectorValue: 'US'
+  },
+//   {
```
Comment (Kuzirashi) [inline app/stores/LanguageStore.js]: Unnecessary comment.

File: app/scss/style.scss (hunk 4 above)
Comment (kennym) [reply app/scss/style.scss]: Absolutely. Thanks for spotting that.

**Comment by kennym** [comment]
I don't have time to implement - feel free to implement this yourselves.

**Pull request closed without merging**